spec/controllers/pockets_controller_spec.rb

```diff
@@ -196,59 +195,72 @@ def edit_weight_call(pocket_id, weight, token)
+
+        it 'does change the state to Weighed' do
+          expect(json_response[:state]).to eq 'Weighed'
+        end
+
+        it 'does add the weight' do
+          expect(json_response[:weight]).to eq weight
+        end
+
+        it 'does return ok' do
+          expect(response).to have_http_status(:ok)
+        end
+
+        it 'does return the pocket as specified in the serializer' do
+          expect(json_response).to eq serializer.new(unweighed_pocket.reload).as_json
+        end
       end
-    end
 
-    context 'when ApiKey is missing' do
-      it 'does return invalid token' do
-        put :edit_weight, params: { id: unweighed_pocket.id, weight: 15 }
-        expect(response).to have_http_status(401)
+      context 'when pocket id is invalid' do
+        it 'does return not found' do
+          add_weight_call(Pocket.pluck(:id).max + 1, weight)
+          expect(response).to have_http_status(:not_found)
+        end
       end
-    end
 
-    context 'when pocket id is invalid' do
-      it 'does return not found' do
-        add_weight_call(Pocket.pluck(:id).max + 1, 12.3, device.auth_token)
-        expect(response).to have_http_status(:not_found)
+      context 'when the weight is invalid or missing' do
+        it 'does return bad request, negative weight' do
+          add_weight_call(unweighed_pocket.id, -14)
+          expect(response).to have_http_status(400)
+        end
+
+        it 'does return bad request, empty weight' do
+          add_weight_call(unweighed_pocket.id, '')
+          expect(response).to have_http_status(400)
+        end
+
+        it 'does return bad request, nil weight' do
+          add_weight_call(unweighed_pocket.id, nil)
+          expect(response).to have_http_status(400)
+        end
+
+        it 'does return bad request, weighed pocket' do
+          add_weight_call(weighed_pocket.id, weight)
+          expect(response).to have_http_status(400)
+        end
       end
     end
 
-    context 'when the weight is invalid or missing' do
-      it 'does return bad request, negative weight' do
-        add_weight_call(unweighed_pocket.id, -14, device.auth_token)
-        expect(response).to have_http_status(400)
-      end
-      it 'does return bad request, empty weight' do
-        add_weight_call(unweighed_pocket.id, '', device.auth_token)
-        expect(response).to have_http_status(400)
-      end
-      it 'does return bad request, nil weight' do
-        add_weight_call(unweighed_pocket.id, nil, device.auth_token)
-        expect(response).to have_http_status(400)
-      end
-      it 'does return bad request, weighed pocket' do
-        add_weight_call(weighed_pocket.id, 80.8, device.auth_token)
-        expect(response).to have_http_status(400)
+    context 'when user is not authenticated' do
+      let!(:user) { create(:user, organization: organization) }
+
+      before(:each) { add_weight_call(unweighed_pocket.id, weight) }
+
+      it 'does return invalid token' do
```

Comment:
Check the error code is the correct one

Reply:
The code is correct because it is in the context where the user is not authenticated. It would be better to change the description to: does return unauthorized, invalid auth token?